pkgs/development/tools/misc/segger-jlink/default.nix

```diff
@@ -0,0 +1,160 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, udev
+, config
+, acceptLicense ? config.segger-jlink.acceptLicense or false
+, fontconfig
+, xorg
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "x86_64";
+      sha256 = "cb43e8656cc2d7c7b42a5f978d065614378e0ef388ef70321ef18c1082450e6a";
+    };
+    i686-linux = {
+      name = "i386";
+      sha256 = "7969354639d8251a7b341bb281649836778bb2ed61166a23779c8db7edc62f54";
+    };
+    aarch64-linux = {
+      name = "arm64";
+      sha256 = "9c1126fae4678df4d6a8e3ee5a1784366efa12050906c49fa95906eb0578ef1d";
+    };
+    armv7l-linux = {
+      name = "arm";
+      sha256 = "e2608cd2f1b666f9e93609e3aa3f6559c7d6f3d9097e84d439c982b23018b1a1";
+    };
+  };
+
+  platform = supported.${stdenv.system} or (throw "unsupported platform ${stdenv.system}");
+
+  version = "792e";
+
+  url = "https://www.segger.com/downloads/jlink/JLink_Linux_V${version}_${platform.name}.tgz";
+
+  src =
+    assert !acceptLicense -> throw ''
+      Use of the "SEGGER JLink Software and Documentation pack" requires the
+      acceptance of the following licenses:
+
+        - SEGGER Downloads Terms of Use [1]
+        - SEGGER Software Licensing [2]
+
+      You can express acceptance by setting acceptLicense to true in your
+      configuration. Note that this is not a free license so it requires allowing
+      unfree licenses as well.
+
+      configuration.nix:
+        nixpkgs.config.allowUnfree = true;
+        nixpkgs.config.segger-jlink.acceptLicense = true;
+
+      config.nix:
+        allowUnfree = true;
+        segger-jlink.acceptLicense = true;
+
+      [1]: ${url}
+      [2]: https://www.segger.com/purchase/licensing/
+    '';
+      fetchurl {
+        inherit url;
+        inherit (platform) sha256;
+        curlOpts = "--data accept_license_agreement=accepted";
+      };
+
+  qt4-bundled = stdenv.mkDerivation {
+    pname = "segger-jlink-qt4";
+    inherit src version;
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+    ];
+
+    buildInputs = [
+      stdenv.cc.cc.lib
+      fontconfig
+      xorg.libXrandr
+      xorg.libXfixes
+      xorg.libXcursor
+      xorg.libSM
+      xorg.libICE
+      xorg.libX11
+    ];
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      # Install libraries
+      mkdir -p $out/lib
+      mv libQt* $out/lib
+
+      runHook postInstall
+    '';
+  };
+
+in stdenv.mkDerivation {
+  pname = "segger-jlink";
+  inherit src version;
+
+  # Currently blocked by patchelf bug
+  # https://github.com/NixOS/patchelf/pull/275
+  #runtimeDependencies = [ udev ];
```

Comment:
Isn't this fixed in patchelf 0.13?

Reply:
I think the bug is somewhat unrelated, Udev is loaded late in the runtime of the program (I guess via dlopen()). Patchelf does not pick up this dependency, and adding `udev` to `runtimeDependencies` does not work either. So I kept the current approach. Otherwise this happens:

```
Connecting to J-Link via USB...FAILED: USBBULK: Failed to load libudev.so. Needed for identification of J-Links connected via USB
Cannot connect to J-Link.
```

Reply:
If I recall correctly `runtimeDependencies` only looks for executables. You want `appendRunpaths`

Reply:
Thanks you, added. Learned something new today :)